File: data/code_pairs/fortran/1678.f90
program increment
  implicit none
  integer, dimension(1:10) :: arr
  integer :: i

  ! Initialize the array
  do i = 1, 10
     arr(i) = i
  end do

  ! Use OpenACC to increment each element in parallel
  !$acc data copy(arr)
  !$acc parallel loop
  do i = 1, 10
     arr(i) = arr(i) + 1
  end do
  !$acc end data

  ! Print the results
  do i = 1, 10
     print *, 'arr(', i, ') = ', arr(i)
  end do
end program increment

File: data/code_pairs/cpp/1678.cpp
#include <iostream>
#include <vector>

int main() {
    std::vector<int> arr(10);
    
    // Initialize the array
    for (int i = 0; i < 10; ++i) {
        arr[i] = i + 1; // Matching Fortran's 1-based indexing
    }

    // Use OpenACC to increment each element in parallel
    #pragma acc data copy(arr[0:10])
    #pragma acc parallel loop
    for (int i = 0; i < 10; ++i) {
        arr[i] = arr[i] + 1;
    }

    // Print the results
    for (int i = 0; i < 10; ++i) {
        std::cout << "arr[" << i << "] = " << arr[i] << std::endl;
    }

    return 0;
}